src/Sentry/GlobalSessionManager.cs

```diff
@@ -245,6 +249,13 @@ private void DeletePersistedSession()
 
     private SessionUpdate EndSession(SentrySession session, DateTimeOffset timestamp, SessionEndStatus status)
     {
+        // If we're ending as 'Exited' but he session has a pending 'Unhandled', end as 'Unhandled'
+        if (status == SessionEndStatus.Exited && session.IsMarkedAsPendingUnhandled)
+        {
+            status = SessionEndStatus.Unhandled;
+            _options.LogDebug("Session ended as 'Unhandled' due to pending status.");
```

Comment:
Nit: just thinking of our future selves seeing this code in 12 month's time... maybe this is clearer to understand? Feel free to ignore if you disagree. I'm guessing this functionality will mostly be used by Unity users.

```suggestion
            _options.LogDebug("Session ended with pending 'Unhandled' (but not `Terminal`) exception.");
```